modules/runner/src/main/ts/ui/Actions.ts

```diff
@@ -0,0 +1,28 @@
+import { makeUrl } from '../core/Utils';
+
+export interface Actions {
+  restartTests: () => void;
+  retryTest: (offset: number, failed: number, retry?: number) => void;
+  skipTest: (offset: number, failed: number) => void;
+  reloadPage: (offset: number, failed: number, retry?: number) => void;
+  updateHistory: (offset: number, failed: number, retry?: number) => void;
+}
+
+export const Actions = (session: string): Actions => {
+  const reloadPage = (offset: number, failed: number, retry = 0) => {
+    const url = makeUrl(session, offset, failed, retry);
+    window.location.assign(url);
+  };
+
+  const updateHistory = (offset: number, failed: number, retry = 0) => {
+    window.history.pushState({}, '', makeUrl(session, offset, failed, retry));
```

Comment:
Don't change it, but just curious if there is a reason you inlined this one but not the above one? Is there some particular heuristic you use?

Reply:
I just copied it across and that's how it was previously :shrug:

Reply:
Fair enough.

Reply:
Fixed anyways :+1: